esx_service/vmdk_ops.py

```diff
@@ -854,11 +854,14 @@ def findDeviceByPath(vmdk_path, vm):
         # to match with the given volume name.
         # Filename format is as follows:
         #   "[<datastore name>] <parent-directory>/tenant/<vmdk-descriptor-name>"
-        logging.debug("d.backing.fileName %s", d.backing.fileName)
-        backing_disk = d.backing.fileName.split(" ")[1]
+        # Assume there is one space between these two parts "[<datastore name>]" and
+        # <parent-directory>/tenant/<vmdk-descriptor-name>
+        logging.debug("findDeviceByPath: d.backing.fileName %s", d.backing.fileName)
+        backing_disk = d.backing.fileName.rsplit(" ", 1)[1]
```

Comment:
That was the point - please check it :-). Directory names (i.e. tenant) and volume names (i.e. files) will accept spaces unless we block them on creation. So please check if we do block it on creation.  I agree with @shaominchen that it's better to block spaces in tenant names and volume names. on creation.
With that said, i still think the code I suggested is slightly more resilient than looking for spaces.